tests/unit/test_step.py

```diff
@@ -141,37 +147,190 @@ def test_composite_step(default_config_params) -> None:
+            },
+        ),
+    ]
+    assert len(subgraph.edges) == len(expected_edges)
+    for edge in expected_edges:
+        assert edge in subgraph.edges(data=True)
+
+
+def test_hierarchical_step() -> None:
+    params = {
+        "name": "step_1",
+        "input_slots": [
+            InputSlot(
+                "step_1_main_input",
+                "DUMMY_CONTAINER_MAIN_INPUT_FILE_PATHS",
+                validate_input_file_dummy,
+            )
+        ],
+        "output_slots": [OutputSlot("step_1_main_output")],
+        "nodes": [
+            BasicStep(
+                "step_1a",
+                input_slots=[
+                    InputSlot(
+                        "step_1a_main_input",
+                        "DUMMY_CONTAINER_MAIN_INPUT_FILE_PATHS",
+                        validate_input_file_dummy,
+                    )
+                ],
+                output_slots=[OutputSlot("step_1a_main_output")],
+            ),
+            BasicStep(
+                "step_1b",
+                input_slots=[
+                    InputSlot(
+                        "step_1b_main_input",
+                        "DUMMY_CONTAINER_MAIN_INPUT_FILE_PATHS",
+                        validate_input_file_dummy,
+                    )
+                ],
+                output_slots=[OutputSlot("step_1b_main_output")],
+            ),
+        ],
+        "edges": [Edge("step_1a", "step_1b", "step_1a_main_output", "step_1b_main_input")],
+        "slot_mappings": {
+            "input": [
+                SlotMapping(
+                    "input", "step_1", "step_1_main_input", "step_1a", "step_1a_main_input"
+                )
+            ],
+            "output": [
+                SlotMapping(
+                    "output", "step_1", "step_1_main_output", "step_1b", "step_1b_main_output"
+                )
+            ],
         },
     }
-
-    assert set(subgraph.edges()) == expected_edges.keys()
-    for source, sink, edge_attrs in subgraph.edges(data=True):
-        assert (
-            edge_attrs["input_slot"].name == expected_edges[(source, sink)]["input_slot_name"]
-        )
-        assert edge_attrs["input_slot"].env_var == expected_edges[(source, sink)]["env_var"]
-        assert (
-            edge_attrs["input_slot"].validator == expected_edges[(source, sink)]["validator"]
-        )
-        assert (
-            edge_attrs["output_slot"].name
-            == expected_edges[(source, sink)]["output_slot_name"]
+    step = HierarchicalStep(**params)
+    assert step.name == "step_1"
+    assert step.input_slots == {
+        "step_1_main_input": InputSlot(
+            "step_1_main_input",
+            "DUMMY_CONTAINER_MAIN_INPUT_FILE_PATHS",
+            validate_input_file_dummy,
         )
+    }
+    assert step.output_slots == {"step_1_main_output": OutputSlot("step_1_main_output")}
+
+    # Test update_implementation_graph for single implementation
```

Comment:
Should these "test update_implementation_graph" sections all be in their own unit tests? These tests are getting long and this seems kinda different than all the stuff above it